Add Watchdog Manager tests

fkjagodzinski opened the pull request:
### Description

<!-- 
    Required
    Add here detailed changes summary, testing results, dependencies 
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->

After the Watchdog driver update from ~#9243~ #10066 the greentea tests also need an update:
* remove `tests-mbed_drivers-watchdog*`,
* add `tests-mbed_platform-watchdog_mgr_reset`.

Please note, **this PR needs #10066** merged first to work.

### Pull request type

<!-- 
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [x] Test update
    [ ] Breaking change

### Reviewers

<!-- 
    Optional
    Request additional reviewers with @username
-->

@c1728p9 @rajkan01

fkjagodzinski committed:
Tests: Watchdog: Remove outdated driver tests

The Watchdog driver has been updated making these tests incompatible
with the new API. The Watchdog driver functionality is tested with the
unit tests located in UNITTESTS/drivers/Watchdog/ .

fkjagodzinski committed:
Tests: Watchdog: Add Watchdog Manager tests

ciarmcom commented:
@fkjagodzinski, thank you for your changes.
@c1728p9 @rajkan01 @ARMmbed/mbed-os-maintainers please review.

fkjagodzinski committed:
Tests: ResetReason: Use the HAL watchdog API

The Watchdog driver has been updated making this test incompatible
with the new API. Use HAL API instead.
